### ISSUE TITLE
Review Checker's Show products recommended toggle is not programmatically labeled

### ISSUE THREAD

**Issue by anna-yeddi** [mozilla-mobile/firefox-ios#17767]
## Steps to reproduce
1. Open Firefox, ensure VoiceOver is running
2. Navigate to any amazon product page and open the Review Checker
3. Navigate to and expand the "Settings" card
4. Use logical VoiceOver navigation gestures (one finger swipes right - forward, our left - backward) to navigate to the toggle for "Show products recommended by Firefox" and observe the VO behavior

### Expected behavior
1. VO focuses the toggle and its on-screen label as one element and it is announced as "{label}, Switch button, {on/off}". 
They should be focused as one element - working example is Settings > General > Automatic updates (attached). Or, if this is not possible, do not focus the label text (not importantForAccessibility) and ensure the accessibleLabel is provided for the toggle that is exactly the same as the on-screen text, so only the toggle is focused with the VO to avoid extra swipes and redundant label announcement

### Actual behavior
1. VO focuses the on-screen label separately that is announced as a static text, and then the next element in the VO focus order is the toggle, that is announced as an unlabeled control: "Switch button, {on/off}"

### Device & build information
* Device: iPhone 12pro, iOS 17.1.2
* Build version: Fx Beta 9000 (36796)
### Notes
Attachments: 
1. Label focused and announced by VO as a static text 
![toggle-label](https://github.com/mozilla-mobile/firefox-ios/assets/40773610/f4f07a8c-719b-42df-b0d7-d581dd690e61)
2. Toggle is focused and announced separately with VO without any label (no indication of it's purpose)
![toggle-control](https://github.com/mozilla-mobile/firefox-ios/assets/40773610/5e61b3ae-4e05-41ea-a1ae-cfb160284d29)
3. Working example of the labeled toggle
![Working example](https://github.com/mozilla-mobile/firefox-ios/assets/40773610/25bd72df-dce6-4106-89fd-7f3868406e1d)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-7971)

**Comment by data-sync-user** [comment]
➤ Andrei Bodea commented:

Verified as fixed on v123 (38491) with iPhone 15 Pro (17.1.2).
